### DISCUSSION TITLE
12/20 팀 주간 회의

### DISCUSSION THREAD

**Discussion by lv0314** [over-ui/unstyled#31] [General]
## 📋 Cherry-pick 사용
- 체리픽을 사용하여 배포하기로 하였으나 다시 한 번 논의
- 단점 : 커밋을 하나하나 찝어줘야 하기 때문에 귀찮을 수 있음
- 장점 : over-ui를 모노 레포 방식으로 배포하기 때문에 커밋 내용을 모아 컴포넌트 별로 배포하기 용이

-> 조금 귀찮아 질 수 있지만 체리픽을 사용하는 것이 모노 레포 배포에 용이하기 때문에 `현재 방식 유지`하기로 결정

## 📋 Merge 방식 - squash vs rebase
- 둘다 선호하는 방식이 없고 프로젝트 내에서 크게 문제될 게 없어 보여서, 보기에 깔끔해 보이는 squash merge를 사용하기로 결정했음
- 오터의 현업 친구분의 조언 : 웬만하면 rebase and merge로 하는 게 좋고, 체리픽을 사용할거라면 squash는 커밋을 합치기 때문에 커밋 번호 찾기가 힘들어질 수 있음

-> 체리픽을 사용하기로 결정했으므로 squash보다는 rebase가 더 적합할 것 같기 때문에 `rebase and merge 방식으로 변경`
## 📋 Poly 컴포넌트 네이밍 컨벤션
- Poly를 사용하여 컴포넌트를 생성할 때 내부에서 껍데기 역할을 해주는 컴포넌트의 네이밍을 초기부터 맞추는게 낫다고 판단

ex)
```
const Demo: Poly.Component<typeof DEFAULT_DEMO_TAG, DemoProps> = React.forwardRef((
    props, forwardedRef
  ) => {
    const { as, children, ...restProps } = props;

    const Component = as || DEFAULT_DEMO_TAG;

    return (
      <Component {...restProps} ref={forwardedRef}>
        {children}
      </Component>
    );
  }
);
``` 
위와 같은 코드가 있을 경우 Component라는 네이밍은 애매함
그렇다고 <Demo />로 만드려는 컴포넌트와 같은 이름을 쓰면 헷갈리기도 하고 쓰기 좀 꺼림칙함

-> 결국 저 부분이 tag를 나타내는 것이기 때문에, `<Tag />`를 사용하기로 결정
## 📋 Playground 관련
- playground는 만들어 놓은 컴포넌트를 화면 상으로 볼 수 있게 해줌
- 테스트 해본 코드를 같이 올려야 할 지, 빼고 올려야 할 지에 대한 논의

-> playground는 usage 페이지로 사용할 수 있을 것 같음
-> 각자 로컬에 코드를 보관하거나, README에 작성을 해두고 추후에 usage 페이지를 만드는 데 이용하기로 결정

**Comment by otterp012** [comment]
정리 고생하셨습니다!
